src/dmi.rs

```diff
@@ -63,3 +68,32 @@ fn create_png(path: &str, width: &str, height: &str, data: &str) -> Result<()> {
     let mut writer = encoder.write_header()?;
     Ok(writer.write_image_data(&result)?)
 }
+
+#[derive(Debug)]
+enum ResizePngError {
+    StringParse(std::num::ParseIntError),
+    Image(ImageError),
+}
+
+impl From<std::num::ParseIntError> for ResizePngError {
+    fn from(error: std::num::ParseIntError) -> Self {
+        ResizePngError::StringParse(error)
+    }
+}
+
+impl From<ImageError> for ResizePngError {
+    fn from(error: ImageError) -> Self {
+        ResizePngError::Image(error)
+    }
+}
+
+fn resize_png(path: &str, width: &str, height: &str) -> std::result::Result<(), ResizePngError> {
```

Comment:
```suggestion
fn resize_png<P: AsRef<Path>>(path: P, width: u32, height: u32) -> std::result::Result<(), Error> {
```

Reply:
Can BYOND call functions with u32 arguments? I thought it explicitly only worked with strings

Reply:
Correct, but you'd convert them to u32 in the `byond_fn!`

Reply:
Trying to do that results in the `?` no longer working for the ParseIntErrors because byond_fn! doesn't actually return a Result

Reply:
That's probably not exactly why it doesn't work, but I have no idea how to make it stop yelling at me with this
![https://i.tigercat2000.net/2020/10/Code_sYiIJS9DzI.png](https://i.tigercat2000.net/2020/10/Code_sYiIJS9DzI.png)